### PR TITLE
DATAES-479 - Allow specifying a HighlightBuilder when creating a query.

### PR DESCRIPTION
**Pull request by jnizet**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

File: src/main/java/org/springframework/data/elasticsearch/core/query/NativeSearchQueryBuilder.java [hunk 1]
```diff
@@ -46,7 +47,8 @@
 	private List<SortBuilder> sortBuilders = new ArrayList<>();
 	private List<FacetRequest> facetRequests = new ArrayList<>();
 	private List<AbstractAggregationBuilder> aggregationBuilders = new ArrayList<>();
-	private HighlightBuilder.Field[] highlightFields;
+    private HighlightBuilder highlightBuilder;
+    private HighlightBuilder.Field[] highlightFields;
```
Comment (saschawx28) [inline src/main/java/org/springframework/data/elasticsearch/core/query/NativeSearchQueryBuilder.java]: wrong indentation

Comment (xhaggi) [inline src/main/java/org/springframework/data/elasticsearch/core/query/NativeSearchQueryBuilder.java]: wrong identation

File: src/main/java/org/springframework/data/elasticsearch/core/query/NativeSearchQueryBuilder.java [hunk 2]
```diff
@@ -90,6 +92,11 @@ public NativeSearchQueryBuilder withFacet(FacetRequest facetRequest) {
 		return this;
 	}
 
+    public NativeSearchQueryBuilder withHighlightBuilder(HighlightBuilder highlightBuilder) {
+        this.highlightBuilder = highlightBuilder;
+        return this;
+    }
```
Comment (xhaggi) [inline src/main/java/org/springframework/data/elasticsearch/core/query/NativeSearchQueryBuilder.java]: same here, wrong identation

**Comment by jnizet** [comment]
@xhaggi sorry for the indentation problems. It should be fixed now.

**Comment by xhaggi** [comment]
Merged, thank you for your contribution.

**Pull request closed without merging**